docs/proposals/07-shoot-control-plane-migration.md

```diff
@@ -0,0 +1,244 @@
+
+Extensions which do not require state migration should set `status.state=nil` in their Custom Resources and trigger a normal reconciliation operation if the CR contains the `core.gardener.cloud/operation=restore` annotation.
+
+Similar to the contract for the [reconcile operation](https://github.com/gardener/gardener/blob/master/docs/extensions/reconcile-trigger.md), the extension controller has to remove the `restore` annotation after the restoration operation has finished.
+
+An additional annotation `gardener.cloud/operation=migrate` is added to the Custom Resources. It is used to tell the extension controllers in the __Source Seed__ that they must stop reconciling resources (in case they are requeued due to errors) and should perform cleanup activities in the Shoot's control plane. These cleanup activities involve removing the finalizers on Custom Resources and deleting them without actually deleting any infrastructure resources.
+
+**Note:** The same size limitations from the previous section are relevant here as well.
+
+### Shoot reconciliation flow changes
+
+The only data which must be stored in the `ShootSnapshot` by the gardenlet is secrets (e.g ca for the API server). Therefore the `botanist.DeploySecrets` step is changed. It is split into two functions which take a list of secrets that have to be generated.
+- `botanist.GenerateSecretState` Generates certificate authorities and other secrets which have to be persisted in the ShootSnapshot and must not be regenerated on the __Destination Seed__.
+- `botanist.DeploySecrets` Takes secret data from the `ShootSnapshot`, generates new ones (e.g. client tls certificates from the saved certificate authorities) and deploys everything in the Shoot's control plane on the __Destination Seed__
+
+
+### ShootSnapshot synchronization controller
+
+The ShootSnapshot synchronization controller will become part of the gardenlet. It syncs the state of extension custom resources from the shoot namespace to the garden cluster and updates the corresponding `spec.extension.state` field in the `ShootSnapshot` resource. The controller can `watch` Custom Resources used by the extensions and update the `ShootSnapshot` only when changes occur.
+
+
+### Migration workflow
+1. Starting migration
+    - Migration can only be started after a Shoot cluster has been successfully created so that the `status.seed` field in the `Shoot` resource has been set
+    - The `Shoot` resource's field `spec.seedName="new-seed"` is edited to hold the name of the __Destination Seed__ and reconciliation is automatically triggered
+    - The Garden Controller Manager checks if the equality between `spec.seedName` and `status.seed`, detects that they are different and triggers migration.
+1. The Garden Controller Manager waits for the __Destination Seed__ to be ready
+1. Shoot's API server is stopped
+1. Backup the Shoot's ETCD.
+1. Extension resources in the __Source Seed__ are annotated with `gardener.cloud/operation=migrate`
```

Comment:
I think the GEP was written before the introduction of the gardenlet and not updated properly. cc @plkokanov 
Basically, GCM is doing nothing because it might not have access to the seeds. The whole control plane migration flow is in the shoot controller which was moved to the gardenlet.